crates/consensus/src/transaction/envelope.rs

```diff
@@ -448,19 +448,19 @@ impl OpTxEnvelope {
     }
 
     /// Returns the inner transaction hash.
-    pub fn hash(&self) -> &B256 {
+    pub fn hash(&self) -> B256 {
```

Comment:
I need this to return a reference.

hash fns need a cleanup, but this change doesn't belong in this pr

Reply:
Hrm ci was failing

Reply:
Because `TxDeposit` hash func